packages/repo-tools/src/commands/openapi/test/index.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+import { join } from 'path';
+import chalk from 'chalk';
+import { runner } from '../runner';
+import { YAML_SCHEMA_PATH } from '../constants';
+import { paths as cliPaths } from '../../../lib/paths';
+import { exec } from '../../../lib/exec';
+
+async function test(
+  directoryPath: string,
+  { port }: { port: number },
+  options?: { update?: boolean },
+) {
+  const openapiPath = join(directoryPath, YAML_SCHEMA_PATH);
+  if (!(await fs.pathExists(openapiPath))) {
+    return;
+  }
+  const opticConfigFilePath = join(directoryPath, 'optic.yml');
+  if (!(await fs.pathExists(opticConfigFilePath))) {
+    return;
+  }
+
+  const opticLocation = (
```

Comment:
@freben When you get a chance, one thing that I've been thinking about on top of this one is where to install optic. I think `repo-tools` is the best place for it, so had to add this to get the binary location to then run in the context of other directories. Hopefully, there's a better way to do this...

one kind of weird thing with Optic is they derive the config file based on context and you can't override it with a passed in value, https://github.com/opticdev/optic/issues/2256. Ideally, we could just run `optic` in `repo-tools` and point to each directory, but for now that's a bit of a pipe dream.